### ISSUE TITLE
False positive cache_clear no member with lru_cache decorator without parenthesis

### ISSUE THREAD

**Issue by vinsonlee** [pylint-dev/pylint#8868]
### Bug description

pylint issues a false positive on cache_clear method when the class method is decoratored with lru_cache without parenthesis.

```
"""Module docstring"""
import functools

class TestObject:
    """class docstring"""

    def method(self) -> None:
        """method docstring"""

    @functools.lru_cache
    def cached_method(self) -> None:
        """method docstring"""


t = TestObject()
t.cached_method.cache_clear()
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:16:0: E1101: Method 'cached_method' has no 'cache_clear' member (no-member)
```

### Expected behavior

With parenthesis on lru_cache decorator.
```
"""Module docstring"""
import functools

class TestObject:
    """class docstring"""

    def method(self) -> None:
        """method docstring"""

    @functools.lru_cache()
    def cached_method(self) -> None:
        """method docstring"""


t = TestObject()
t.cached_method.cache_clear()
```

```
% pylint a.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.4
Python 3.11.4 (main, Jun 20 2023, 16:59:59) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```
